arbiter/config.go

```diff
@@ -190,6 +191,20 @@ func (cfg *Config) adjustConfig() error {
 }
 
 func (cfg *Config) configFromFile(path string) error {
-	_, err := toml.DecodeFile(path, cfg)
-	return errors.Trace(err)
+	metaData, err := toml.DecodeFile(path, cfg)
+
+	// If any items in confFile file are not mapped into the Config struct, issue
+	// an error and stop the server from starting.
+	if err != nil {
+		return err
+	}
+	if undecoded := metaData.Undecoded(); len(undecoded) > 0 {
+		var undecodedItems []string
+		for _, item := range undecoded {
+			undecodedItems = append(undecodedItems, item.String())
+		}
+		err = errors.Errorf("arbiter config file %s contained unknown configuration options: %s", path, strings.Join(undecodedItems, ", "))
```

Comment:
Maybe we can create a function in `pkg/util` like `StrictDecodeFile` and then reuse it in the `configFromFile` methods of different packages?

Reply:
Nice suggestion. I will take a try.

Reply:
Great :)

Reply:
Updated.